opendevin/llm/llm.py

```diff
@@ -178,7 +177,31 @@ def wrapper(*args, **kwargs):
 
             # skip if messages is empty (thus debug_message is empty)
             if debug_message:
-                resp = completion_unwrapped(*args, **kwargs)
+                # Check if the model is an Anthropic model
+                is_anthropic_model = 'anthropic' in self.config.model
+
+                if is_anthropic_model:
+                    # Add cache control to the messages
+                    for message in messages:
+                        if 'content' in message and isinstance(
+                            message['content'], list
+                        ):
+                            for content in message['content']:
+                                if isinstance(content, dict) and 'text' in content:
+                                    # Add cache control to the content
+                                    content['cache_control'] = {'type': 'ephemeral'}
```

Comment:
There is a limit of 4 checkpoints, from Anthropic docs. Adding it to all messages doesn't seem to work.

Reply:
Okay will work on this!